### PR TITLE
Highlight selected file in the PR file tree

### PR DESCRIPTION
**Pull request by yusifeng**
before
before

after
after

### CODE REVIEW

**Comment by KN4CK3R** [comment]
Could you add before/after screenshots or explain what it does?

**Comment by yusifeng** [comment]
> Could you add before/after screenshots or explain what it does?



> Could you add before/after screenshots or explain what it does?
I have already add 2 gif in comment

**Comment by silverwind** [comment]
This is nice, it's something I wanted to add as well.

File: web_src/js/components/DiffFileTreeItem.vue [hunk 1]
```diff
@@ -119,6 +127,12 @@ export default {
   border-radius: 4px;
 }
 
+.item.filewrapper.active {
```
Comment (delvh) [inline web_src/js/components/DiffFileTreeItem.vue]: Why not combine the three rules with the same content into one?
What I mean is a rule like
`.item.filewrapper.active, .item.filewrapper:hover, div.directory:hover{`

File: web_src/js/components/DiffFileTreeItem.vue [hunk 2]
```diff
@@ -1,7 +1,7 @@
 <template>
   <div v-show="show" :title="item.name">
     <!--title instead of tooltip above as the tooltip needs too much work with the current methods, i.e. not being loaded or staying open for "too long"-->
-    <div class="item" :class="item.isFile ? 'filewrapper gt-p-1 gt-ac' : ''">
+    <div class="item" :class="[item.isFile ? 'filewrapper gt-p-1 gt-ac' : '', activeHash === genCompleteHash(item.file?.NameHash) ? 'active' : '']">
```
Comment (delvh) [inline web_src/js/components/DiffFileTreeItem.vue]: I think `selected` is a more fitting name than `active` here…

File: web_src/js/components/DiffFileTree.vue [hunk 3]
```diff
@@ -5,7 +5,7 @@
   >
     <!-- only render the tree if we're visible. in many cases this is something that doesn't change very often -->
     <div class="ui list">
-      <DiffFileTreeItem v-for="item in fileTree" :key="item.name" :item="item" />
+      <DiffFileTreeItem v-for="item in fileTree" :key="item.name" :item="item" :active-hash="activeHash"/>
```
Comment (delvh) [inline web_src/js/components/DiffFileTree.vue]: Same here, what about calling it `selectedFile` instead?

**Comment by wxiaoguang** [comment]
It seems that the "current selected item" is a global state, maybe we should consider about using https://vuex.vuejs.org/

> However, the simplicity quickly breaks down when we have multiple components that share a common state:

File: web_src/js/components/DiffFileTreeItem.vue (hunk 1 above)
Comment (yusifeng) [reply web_src/js/components/DiffFileTreeItem.vue]: done!

File: web_src/js/components/DiffFileTree.vue (hunk 3 above)
Comment (yusifeng) [reply web_src/js/components/DiffFileTree.vue]: done!

**Comment by silverwind** [comment]
> It seems that the "current selected item" is a global state

You mean shared between the two components? Yes I guess some state management across components would be nice, but I have no experience with vue global state. In React, I use [zustand](https://github.com/pmndrs/zustand), so something similarily simple would be great.

**Comment by yusifeng** [comment]
> This is nice, it's something I wanted to add as well.

Compared to other similar tools（like bitbucket）, Gitea render too many file in this page, especially when there are many files changed in a pull request, it may cause the page to lag.

I am prepare to render one file only other files will auto close , like Collapse Compoent.
20230406184013_rec_

**Comment by lunny** [comment]
> > This is nice, it's something I wanted to add as well.
> 
> Compared to other similar tools（like bitbucket）, Gitea render too many file in this page, especially when there are many files changed in a pull request, it may cause the page to lag.
> 
> I am prepare to render one file only other files will auto close , like Collapse Compoent. 20230406184013_rec_ [ 20230406184013_rec_ ](https://user-images.githubusercontent.com/36984894/230353931-c92a2def-dcd3-4004-ab7e-b4aa723b6e42.gif) [ ](https://user-images.githubusercontent.com/36984894/230353931-c92a2def-dcd3-4004-ab7e-b4aa723b6e42.gif)

Please do that in another PR, one PR do one thing.

**Comment by silverwind** [comment]
> Compared to other similar tools（like bitbucket）, Gitea render too many file in this page, especially when there are many files changed in a pull request, it may cause the page to lag.

We have an open issue around for Diff performance with many files. Yes, it's for another PR, not this one.

**Review by delvh — APPROVED** [review]

**Review by silverwind — CHANGES_REQUESTED** [review]
Please use `--color-active` for `.selected` and keep `--color-hover` for `:hover`.

**Comment by yusifeng** [comment]
> 

done!

**Review by silverwind — APPROVED** [review]

**Comment by jolheiser** [comment]
🎺 🤖

**Pull request merged**